src/notifications/send_to_mattermost.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash -eux
+#
+# Send custom notifications to a Mattermost channel.
+#
+
+set -o pipefail
+
+# If not given as an env, script will fail
+MM_CHANNEL_ID=${MM_CHANNEL_ID:-}
+MM_BOT_TOKEN=${MM_BOT_TOKEN:-}
+MM_SERVER=${MM_SERVER:-}
+
+if [ -z "$MM_CHANNEL_ID" ] || [ -z "$MM_BOT_TOKEN" ] || [ -z "$MM_SERVER" ]
+then
+    echo "ERROR: \$MM_CHANNEL_ID, \$MM_BOT_TOKEN and \$MM_SERVER must be provided"
+    exit 1
+fi
+
+# The message payload
+PAYLOAD_FILE="${PWD}/payload.json"
+
+# # What's the status of the job/pipeline that triggered this script?
+FINAL_STATUS=${FINAL_STATUS:-"unkown"}
+
+# Set "grey" for unknown states, "red" and "green" for error and success, respectively
+COLOR="#B0B0B0"
+if [[ "${FINAL_STATUS}" == "success" ]]
+then
+    COLOR="#33CC33"
+elif [[ "${FINAL_STATUS}" == "fail"* ]]
+then
+    COLOR="#CC0000"
+fi
+
+cat>${PAYLOAD_FILE} <<EOF
+{
+    "channel_id": "${MM_CHANNEL_ID}",
+    "props": {
+        "attachments": [
+            {
+                "fallback": "${SUMMARY}",
+                "text": "${SUMMARY}",
+                "color": "${COLOR}",
+                "title": "${TITLE}", 
+                "title_link": "${URL}",
+                "footer": "${FOOTER}"
+            }
+        ]
+    }
+}
+EOF
+
+echo "Posting message to Mattermost channel ${MM_CHANNEL_ID}"
```

Comment:
to the Mattermost channel no?

Reply:
both are acceptable, but what's actually missing is `'s`